drivers/flash/flash_stm32_ospi.c

```diff
@@ -319,21 +320,18 @@ static int stm32_ospi_mem_ready(OSPI_HandleTypeDef *hospi, uint8_t nor_mode, uin
 	/* Configure automatic polling mode command to wait for memory ready */
 	if (nor_mode == OSPI_OPI_MODE) {
 		s_command.Instruction = SPI_NOR_OCMD_RDSR;
+		s_command.AddressMode = HAL_OSPI_ADDRESS_8_LINES;
+		s_command.DummyCycles = (nor_rate == OSPI_DTR_TRANSFER)
+					? SPI_NOR_DUMMY_REG_OCTAL_DTR
+					: SPI_NOR_DUMMY_REG_OCTAL;
 	} else {
 		s_command.Instruction = SPI_NOR_CMD_RDSR;
-		s_command.InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE;
-		s_command.DataMode = HAL_OSPI_DATA_1_LINE;
```

Comment:
Many dual and quad mode ICs use 1 data line mode for that. 
By removing that line a data line mode is enforced to be 2 or 4 lines in the prepare cmd, but shall be 1 line.

Reply:
Forcing the `s_command.DataMode = HAL_OSPI_DATA_1_LINE` for all non-ospi modes
after the prepare command which sets the  s_command.DataMode

Reply:
> Do you mean s_command.InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE too ?

Sorry, yes.